rpcserver.go

```diff
@@ -1103,3 +1108,72 @@ func (r *rpcServer) SendAsset(ctx context.Context,
 		TotalFeeSats: int64(resp.TotalFees),
 	}, nil
 }
+
+// SubscribeSendAssetEventNtfns registers a subscription to the event
+// notification stream which relates to the asset sending process.
+func (r *rpcServer) SubscribeSendAssetEventNtfns(
+	in *tarorpc.SubscribeSendAssetEventNtfnsRequest,
+	ntfnStream tarorpc.Taro_SubscribeSendAssetEventNtfnsServer) error {
+
+	eventSubscriber := chanutils.NewEventReceiver[tarofreighter.Event](
+		chanutils.DefaultQueueSize,
+	)
+	err := r.cfg.ChainPorter.RegisterSubscriber(eventSubscriber, false, false)
+	if err != nil {
+		return err
+	}
+
+	// This goroutine will read from the ChainPorter event subscription and
+	// forward to the RPC stream.
+	errorsChan := make(chan error, 1)
+	go func() {
+		for {
+			select {
+			case event := <-eventSubscriber.NewItemCreated.ChanOut():
+
+				rpcEvent, err := marshallSendAssetEvent(event)
+				if err != nil {
+					err := fmt.Errorf("failed to marshall"+
+						"ChainPorter event into RPC "+
+						"event: %w", err)
+					errorsChan <- err
+					return
+				}
+
+				err = ntfnStream.Send(rpcEvent)
+				if err != nil {
+					err := fmt.Errorf("failed to RPC "+
+						"stream send event: %w", err)
+					errorsChan <- err
+					return
+				}
+			}
+		}
+	}()
+
+	// Handle errors from the goroutine.
+	if err := <-errorsChan; err != nil {
```

Comment:
This should also select on the main quit statement.

Reply:
I've added
```
case <-r.quit:
    return
```
to the goroutine's switch statement in the latest commit. Is that the correct change?

Reply:
I've change the error handling statement to be:
```
	select {
	// Handle errors from the goroutine.
	case err := <-errorsChan:
		if err != nil {
			return fmt.Errorf("error forwarding send asset event "+
				"notifications to RPC stream send: %w", err)
		}
	case <-r.quit:
		return nil
	}
```

Should the `case <-r.quit` raise an error? I think I've seen that in other places in the code. But I'm not sure if it makes sense to do that here. Does it?

Reply:
The `errorsChan` and related setup has been removed in the latest commits.